### PR TITLE
Multivcs trunk

### PR DESCRIPTION
**Pull request by peterdeme**
## Description of the change

This is a PR for the multi VCS work which includes all VCS integrations.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (non-breaking change that adds documentation)

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development
- [x] Examples for new resources and data sources have been added
- [x] Default values have been documented in the description (e.g., "Dummy: (Boolean) Blah blah. Defaults to `false`.)
- [x] If the action fails that checks the documentation: Run `go generate` to make sure the docs are up to date

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] Pull Request is no longer marked as "draft"
- [ ] Reviewers have been assigned
- [ ] Changes have been reviewed by at least one other engineer

### CODE REVIEW

File: spacelift/data_stack.go [hunk 1]
```diff
@@ -219,10 +234,15 @@ func dataStack() *schema.Resource {
...
+						"id": {
+							Type:        schema.TypeString,
+							Description: "ID of the GitHub Enterprise application",
+							Computed:    true,
+						},
 						"namespace": {
 							Type:        schema.TypeString,
 							Description: "GitHub Enterprise namespace of the stack's repository",
-							Required:    true,
+							Computed:    true,
```
Comment (truszkowski) [inline spacelift/data_stack.go:245]: `namespace` shouldn't be required?

File: spacelift/data_stack.go [hunk 2]
```diff
@@ -103,6 +103,11 @@ func dataStack() *schema.Resource {
 				Computed:    true,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeString,
+							Description: "ID of the Azure Devops project",
```
Comment (truszkowski) [inline spacelift/data_stack.go:108]: I think it should be "ID of the Azure DevOps integration"

File: spacelift/data_stack.go [hunk 3]
```diff
@@ -219,10 +234,15 @@ func dataStack() *schema.Resource {
 				Computed:    true,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeString,
+							Description: "ID of the GitHub Enterprise application",
```
Comment (truszkowski) [inline spacelift/data_stack.go:239]: I think it should be "ID of the GitHub Enterprise integration".

File: spacelift/data_module.go [hunk 4]
```diff
@@ -93,6 +108,11 @@ func dataModule() *schema.Resource {
 							Description: "GitHub Enterprise namespace of the stack's repository",
 							Required:    true,
 						},
+						"id": {
+							Type:        schema.TypeString,
+							Description: "ID of the GitHub Enterprise application",
```
Comment (truszkowski) [inline spacelift/data_module.go:113]: I think it should be "ID of the GitHub Enterprise integration" instead of "application"

**Review by truszkowski — DISMISSED** [review]

File: spacelift/data_stack.go (hunk 3 above)
Comment (peterdeme) [reply spacelift/data_stack.go:239]: true

File: spacelift/data_stack.go (hunk 1 above)
Comment (peterdeme) [reply spacelift/data_stack.go:245]: no, because this is a data source, not a resource.

**Comment by peterdeme** [comment]
I'm squashing the commits.

File: spacelift/data_stack.go (hunk 1 above)
Comment (truszkowski) [reply spacelift/data_stack.go:245]: for bitbucket cloud, bitbucket datacenter and azure devops we have required=true as well.

Comment (truszkowski) [reply spacelift/data_stack.go:245]: And in `data_module.go`, the same.

Comment (peterdeme) [reply spacelift/data_stack.go:245]: strange, we'll look into it

**Review by truszkowski — APPROVED** [review]

**Pull request merged**